custom/enikshay/tasks.py

```diff
@@ -179,6 +182,24 @@ def is_dose_taken(cases):
 
         return dose_taken_by_date
 
+    def get_adherence_schedule_start_date(self):
+        # return property 'adherence_schedule_date_start' of episode case, cast to datetime
+        raw_date = self.get_property('adherence_schedule_date_start')
```

Comment:
I know you didn't change this here, but I think every call to `dynamic_case_properties` does a DB lookup. You might want to cache that in some way (e.g. `self.properties`) in the `get_property` method.